test/ClientWidgetApi-test.ts

```diff
@@ -0,0 +1,348 @@
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { waitFor } from '@testing-library/dom';
+import { ClientWidgetApi } from "../src/ClientWidgetApi";
+import { WidgetDriver } from "../src/driver/WidgetDriver";
+import { Capability } from '../src/interfaces/Capabilities';
+import { IRoomEvent } from '../src/interfaces/IRoomEvent';
+import { IWidgetApiRequest } from '../src/interfaces/IWidgetApiRequest';
+import { IReadRelationsFromWidgetActionRequest } from '../src/interfaces/ReadRelationsAction';
+import { ISupportedVersionsActionRequest } from '../src/interfaces/SupportedVersionsAction';
+import { WidgetApiFromWidgetAction } from '../src/interfaces/WidgetApiAction';
+import { WidgetApiDirection } from '../src/interfaces/WidgetApiDirection';
+import { Widget } from '../src/models/Widget';
+import { PostmessageTransport } from '../src/transport/PostmessageTransport';
+
+jest.mock('../src/transport/PostmessageTransport')
+
+afterEach(() => {
+    jest.resetAllMocks();
+})
+
+function createRoomEvent(event: Partial<IRoomEvent> = {}): IRoomEvent {
+    return {
+        type: 'm.room.message',
+        sender: 'user-id',
+        content: {},
+        origin_server_ts: 0,
+        event_id: 'id-0',
+        room_id: '!room-id',
+        unsigned: {},
+        ...event,
+    };
+}
+
+describe('ClientWidgetApi', () => {
+    let capabilities: Capability[];
+    let iframe: HTMLIFrameElement;
+    let driver: jest.Mocked<WidgetDriver>;
+    let clientWidgetApi: ClientWidgetApi;
+    let transport: PostmessageTransport;
+    let emitEvent: Parameters<PostmessageTransport["on"]>["1"];
+
+    async function loadIframe(caps: Capability[] = []) {
+        capabilities = caps;
+
+        const ready = new Promise<void>(resolve => {
+            clientWidgetApi.once('ready', resolve);
+        });
+
+        iframe.dispatchEvent(new Event('load'));
+
+        await ready;
+    }
+
+    beforeEach(() => {
+        capabilities = [];
+        iframe = document.createElement('iframe');
+        document.body.appendChild(iframe);
+
+        driver = {
+            readEventRelations: jest.fn(),
+            validateCapabilities: jest.fn(),
+        } as Partial<WidgetDriver> as jest.Mocked<WidgetDriver>;
+
+        clientWidgetApi = new ClientWidgetApi(
+            new Widget({
+                id: "test",
+                creatorUserId: "@alice:example.org",
+                type: "example",
+                url: "https://example.org",
+            }),
+            iframe,
+            driver,
+        );
+
+        ([transport] = jest.mocked(PostmessageTransport).mock.instances);
```

Comment:
```suggestion
        [transport] = jest.mocked(PostmessageTransport).mock.instances;
```

Reply:
👍 I think I never used this for array destructions into an existing variable and assumed it behaves like when destructuring an object.